src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -232,18 +241,21 @@ public void BeginConnectPeer(String toPeerId, ConnectStatusCallback connectStatu
         }
     }
 
+    //this is always called in context of thread that created instance of the library
```

Comment:
@DrJukka The connected method is, I think, unnecessarily hard to read. Could you please refactor it into two methods, one called ConnectedIncoming and another called ConnectedOutgoing and whomever calls this can then switch between the two based on the incoming value? It would make the code much easier to read.